### PR TITLE
[gree] Minor improvements

### PR DESCRIPTION
**Pull request by markus7017**
This PR adds some improvements
- Improved timeout handling on temporary networks issues
- currentTemperature is updated with NaN when no value is available (device returns -40)
- channel swingUpDown now accepts values 0-11

### CODE REVIEW

File: bundles/org.openhab.binding.gree/README.md
Comment (Hilbrand) [inline bundles/org.openhab.binding.gree/README.md:44]:
```suggestion
| currentTemperature | Number:Temperature | Displays the current room temperature (if supported by unit, otherwise NaN).               |
```

File: bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java
Comment (Hilbrand) [inline bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java:549]: Just hope it will never gets 40 degrees C :wink:

File: bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java
Comment (Hilbrand) [inline bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java:167]: Does it make sense to retry 3 times in a row without some wait time?

**Review by Hilbrand — COMMENTED** [review]
Just some comments.

File: bundles/org.openhab.binding.gree/README.md
Comment (markus7017) [reply bundles/org.openhab.binding.gree/README.md:44]: chaged

File: bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java:549]: The reported temp can never be 0. This would be -40°C, I think there is some room for the implementation :-)

Comment (Hilbrand) [reply bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java:549]: Ah you're right I read that the wrong way..

File: bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java
Comment (markus7017) [reply bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java:167]: There is no delay in submitting commands: 3 retries in a row
only the status poll has delays, which means "unit was not available over the last 3 poll cycles"

I think that makes sense. Nevertheless I changed the code to use a local variable rather than the instance variable used to count the status poll retries.

**Comment by markus7017** [comment]
@Hilbrand done

**Comment by Hilbrand** [comment]
Don't forget to push the changes.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Comment by markus7017** [comment]
Hi @fwolter, could you do the 2nd review?

**Comment by Hilbrand** [comment]
Not directly needed to have a second review on enhancements. I was just waiting for the build.

**Comment by markus7017** [comment]
Thanks @Hilbrand , worked super fast

**Pull request merged**
